activestorage/test/template/image_tag_test.rb

```diff
@@ -0,0 +1,40 @@
+require "test_helper"
+require "database/setup"
+
+class User < ActiveRecord::Base
+  has_one_attached :avatar
+end
+
+class ActiveStorage::ImageTagTest < ActionView::TestCase
+  tests ActionView::Helpers::AssetTagHelper
+
+  setup do
+    @blob = create_image_blob filename: "racecar.jpg"
+  end
+
+  test "blob" do
+    assert_dom_equal %(<img alt="Racecar" src="#{polymorphic_url @blob}" />), image_tag(@blob)
+  end
+
+  test "variant" do
+    variant = @blob.variant(resize: "100x100")
+    assert_dom_equal %(<img alt="Racecar" src="#{polymorphic_url variant}" />), image_tag(variant)
+  end
+
+  test "attachment" do
+    attachment = ActiveStorage::Attachment.new(blob: @blob)
+    assert_dom_equal %(<img alt="Racecar" src="#{polymorphic_url attachment}" />), image_tag(attachment)
+  end
+
+  test "error when attachment's empty" do
+    @user = User.create!(name: "DHH")
+
+    assert_not @user.avatar.attached?
+    assert_raises(ArgumentError) { image_tag(@user.avatar) }
```

Comment:
This now raises `ArgumentError: Can't resolve image into URL: to_model delegated to attachment, but attachment is nil`. Not exactly what I was going for, but good enough I suppose?